src/damavand/cloud/aws/resources/glue_component.py

```diff
@@ -10,19 +10,100 @@
+           * For the Z.2X worker type, each worker maps to 2 M-DPU (8vCPU, 64 GB of m emory, 128 GB disk), and provides up to 8 Ray workers based on the autoscaler.
+    :param enable_glue_datacatalog: To use the Glue catalog as the metadata catalog
+    :param enable_continuous_cloudwatch_log:
+    :param enable_continuous_log_filter: When set to true it reduces the amount of logging.
+        For more information see https://repost.aws/knowledge-center/glue-reduce-cloudwatch-logs
+    :param enable_metrics: Enables observability metrics about the worker nodes.
+    :param enable_observability_metrics: Enables extra Spark-related observability metrics such as how long a tasks takes.
+        This parameter could increase cloud costs significantly.
+    :param script_args: The arguments that your own script consumes {"--arg1": "arg1 value"}
+    :param schedule: A time-based cron-like schedule for the job. For syntax rules see https://docs.aws.amazon.com/glue/latest/dg/monitor-data-warehouse-schedule.html
+    :param job_type: Specify if the job is streaming or batch. Possible values: glueetl, gluestreaming
+    """
+    # Parameters for Pulumi Glue Job
     name: str
-    description: str
-    options: dict[str, Any] = field(default_factory=dict)
+    description: str = None
+    script_location: str = None
+    extra_libraries: list[str] = field(default_factory=list)
+    execution_class: str = "STANDARD"
+    max_concurrent_runs: int = 1,
+    glue_version: str = "4.0"
+    enable_auto_scaling: bool = True
+    max_capacity: int = 5
+    max_retries: int = 0
+    number_of_workers: int = 2
+    tags: dict = None
+    timeout: int = 2880
+    worker_type: str = "G.1X"
+    enable_glue_datacatalog: bool = True
+    enable_continuous_cloudwatch_log:  bool = False
+    enable_continuous_log_filter:  bool = True
+    enable_metrics: bool = False
+    enable_observability_metrics: bool = False
+    script_args: dict = field(default_factory=dict)
+    schedule: str = None
+    job_type: str = "glueetl"
 
 
 @dataclass
 class GlueComponentArgs:
+    """
+    Glue job definitions and infrastructure dependencies such as IAM roles, external connections, code and data storage.
+
+    :param jobs: the list of GlueJobDefinition to deploy
+    :param execution_role: the IAM role attached to the Glue jobs if it exists, if not one will be createdw
+    :param database_name: name of the Glue database if it exists, if not one will be created
+    :param code_repository_bucket_name: name of the s3 code repository database if it exists, if not one will be created
+    :param data_bucket_name: name of the s3 bucket to store data, if it exists, if not one will be created
+    :param kafka_checkpoints_bucket_name: name of the s3 bucket to store checkpoints if it exists, if not one will be created
+    """
```

Comment:
@callarelli93 we should also decide on the docstring standards and use one standard everywhere. The only important thing in choosing the standard is how well the standard is supported with MKDocs.

Reply:
@kkiani In [Inception](https://github.com/DataChefHQ/inception/blob/main/template/%7B%25%20if%20language%20%3D%3D%20'Python'%20%25%7D%7B%7B%20project_name%20%7D%7D%7B%25%20endif%20%25%7D/pyproject.toml.jinja#L82), we agreed to continue with [Google Style](https://google.github.io/styleguide/pyguide.html). I fixed all of the comments to be Google Style compatible.